### ISSUE TITLE
Mypal 68.14.2 gets much slower over time with use than 68.14.1

### ISSUE THREAD

**Issue by K4sum1** [Feodor2/Mypal68#478]
So this seems unrelated to the graphics issue. I upgraded to 68.14.2 to test it again, and after a few hours it got weirdly slow. Freezing and taking multiple seconds to do basic things like paste text on a web page or loading an image. It's also not consistently slow either, after a freeze it can be almost normal for a minute or two before freezing again.

**Comment by Feodor2** [comment]
No details, nothing to do.

I do not confirm this, I use it without restart for weeks.

**Comment by K4sum1** [comment]
I'm not sure how to get detail or how to debug the issue.

**Comment by JimMP68User** [comment]
I do find that 14.2b will continue to eat up memory, and after a few hours of heavy surfing its necessary to close it and start over ... especially watching a series of shorts on YT and having a few YT video tabs open but not playing right at that time also checking Twitter every so often, and a few Rumble videos too ... it was worse when another application already chewed up 1 GB of memory and after an hour 14,2b was slowing up and memory use goes over 4GB on a 4GB physical memory Win Xp Dell 755 computer ... the browser is stable, but it does slow down always eats more memory when surfing to new YT videos. Overall, I'm glad it is available and works on my Xp box.

**Comment by K4sum1** [comment]
Huh, I did notice that the browser content processes crashed more often than usual. I wonder if it's some sort of RAM leak. My memory sucks so I kinda forgot that happened. I had it happen most when using GitHub and searching big code files.

**Comment by Feodor2** [comment]
Okey 4gb - 10 tubes, 10 githubs with large files poked and git blame, 3 twitters poked, 4 hours so far - no errors and slows.

**Comment by JimMP68User** [comment]
> Ah it may be not obvious for you, if something occupies all memory it slows of course

I know; And reloading fixes it, I have learned to live with that. This is just observation. No complaint here. <smile face>

**Comment by K4sum1** [comment]
I have 64 GB of RAM, so paging or running out of RAM isn't really an issue here. If I run out of RAM, the content process usually just crashes, which now that I'm back doing stuff with 68.14.1, hasn't happened once yet.

**Comment by Feodor2** [comment]
I still do not reproduce, I told I poked many github big files along with the youdupe and twitter

If this about the crash you are to give the crash log.

**Comment by K4sum1** [comment]
I really wish you would just make changes commit by commit instead of a single big version commit. I could probably have found the cause by now if I knew what to look at.

**Comment by Feodor2** [comment]
How it is related, at the beginning I try did this, then decided this is a futile too much.

You are to tell details how to reproduce the bug, again may be you set very specific things and forgot this, without this it is nothing to do at all.

**Comment by K4sum1** [comment]
> How it is related

If a content process tries to go over the 32 bit RAM limit, it just crashes. That's what I mean 32 bit related. It would go away if I use 64 bit.

I might just try recording a video of my desktop just me using the browser and wait until it happens.

**Comment by RamonUnch** [comment]
if there is a memory in leak maybe a memory usage report could help? you can get one easily from the `about:memory` tab.

The idea would be to first start mypal and save the report, then surf for a couple hours and when it starts to slow down you close all tabs and go in `about:memory` again, click *minimize memory* usage and see if the memory reduces back to reasonable value. If not, save another report and then you can even look at the diff between two reports.

**Comment by Feodor2** [comment]
>If a content process tries to go over the 32 bit RAM limit, it just crashes.

So this not about 32 limit or what?
Yes this is very specific thing, sure this is not to happen of regular browsing usage and winxp.
Once I did this on intent on win7x64, opened very very bunch of youdubes. I thing is to look how original Firefox handle this, but not today at all, consider this is not important.

**Comment by PLudkovski** [comment]
v68.14.2b is completely unusable at least on NT5.2 (XP64 / Win2003 x64), because of extreme lags.
v68.14.1b is much better, but unfortunately has instant-crashes where all windows will collapse all of a sudden, without any warning, often caused when having open (multiple) Youtube windows.

@Feodor2
If you build it for the old legacy OSes like NT5.1/NT5.2 you should run and test it on them as well.
Makes not too much sense, using a newer OS to build AND test it, when you compile it for an older OS.

**Comment by Ariel-Schnee** [comment]
> v68.14.2b is completely unusable at least on NT5.2 (XP64 / Win2003 x64), because of extreme lags. v68.14.1b is much better, but unfortunately has instant-crashes where all windows will collapse all of a sudden, without any warning, often caused when having open (multiple) Youtube windows.
> 
> @Feodor2 If you build it for the old legacy OSes like NT5.1/NT5.2 you should run and test it on them as well. Makes not too much sense, using a newer OS to build AND test it, when you compile it for an older OS.

You should use 68.13.7b (32-bit) instead. It doesn't have those problems. Seems to be the last version that is stable on XP.

**Comment by K4sum1** [comment]
@Feodor2 If possible, you really should rebuild Mypal68 and make all your changes on top of 68.12.0 into easily identifiable commits. If you do that, I could figure out the cause for new issues caused by a new version or at least narrow it down to a specific commit.

**Comment by Feodor2** [comment]
@PLudkovski 
Yes I don't test on xp64 sorry, don't use it, I thing its very poor compared to winxp, almost useless, better win7 or vista.
So I test on winxp and win7х64 every version, and its running better and less errors than 68.14.1 and 68.14.0.
Seems I already fixed some specific xp64 crash, If this is more problems, then you are to provide details, how much times to tell without details it is nothing to do.

@K4sum1
I told this is too much futile, I wont make a commit on the every fart, I think that the goal to have a browser keep updated quickly.
And I do not ask you to figure out, and narrow it down to a specific commit, this is about just use browser for browsing by usual way and with common system settings.

**Comment by K4sum1** [comment]
Here is a screen recording of me making a commit for r3dfox using Mypal68 68.14.2, and how laggy it is when the issue occurs. Also open is task manager showing a process list and what is going on. It seems like something starts eating a lot of RAM and making the application appear to freeze before being killed as it goes right back down where it was before when the application unfreezes. I also switched to the x64 build to see if it happened there too. Nothing seems at the 32 bit 4GB limit, but also nothing has crashed so idk.

Can't upload directly onto GitHub as it's over 10MB.
https://awau.moe/225DpF3.mp4

**Comment by NS-Clone** [comment]
ну 68.14.2 таки явно систематически уходит в около 100%
чего на квантуме раньше не наблюдалось

**Comment by Feodor2** [comment]
Still do not reproduce, I have 27 tabs and with many githubs and have made commits like you, nothing unusual, all is fine.
What about checking on the clean install?

**Comment by K4sum1** [comment]
The issue takes time to occur. I wouldn't want to use a clean install for long enough for it to occur. However maybe I could rebuild my install with a fraction of the addons and see what happens.

**Comment by Feodor2** [comment]
What time, do I keep using the browser then shall more commit.

**Comment by K4sum1** [comment]
Let me try something and see what happens. Maybe I can find a fast way to reproduce it.

**Comment by K4sum1** [comment]
So I couldn't think of anything, and started researching again how WebRender in Firefox uses SRW locks, and ended with the issue starting to occur. I'm using mostly just GitHub and searching files.

It's a lot less common, I saw it happen once in task manager, but now it's staying calm. Will continue to see it probably happen more.

**Comment by K4sum1** [comment]
Why it was staying calm is that it seems to be limited to one content process. Me typing here did not trigger the issue, and the tab I have this issue open in was still responsive when it was doing the thing in task manager. So limiting to single content process should make it more common and faster to reproduce.

**Comment by NS-Clone** [comment]
из субъективных наблюдений
закрываешь дискорд
жмешь minimize memory usage
и уходишь в 100% до перезапуска браузера
![2024-07-17_184745_](https://github.com/user-attachments/assets/c143536e-304b-45ca-ad8f-ae53909414ca)

**Comment by NS-Clone** [comment]
листал olx
просмотрел страниц максимуум 15
периодически открывая объявления в отдельной вкладке и закрывая обратно
рама вся утекла
клацнул minimize memory usage ушел в 100% на 2 минуты

**Comment by Feodor2** [comment]
20 olx к 20 других сайтов до этого открытым - не утекло.
Та дело даже не в этом, а в том что товарищ типа утверждает что в 14.1 всё хорошо было а в 14.2 уже плохо, но я не понимаю, ничего кроме явы новых функций которые пока ещё в отключенном состоянии и network preload не добавлено, и у 90% остальных нормально тоже

**Comment by NS-Clone** [comment]
> 20 olx к 20 других сайтов до этого открытым - не утекло. Та дело даже не в этом, а в том что товарищ типа утверждает что в 14.1 всё хорошо было а в 14.2 уже плохо, но я не понимаю, ничего кроме явы новых функций которые пока ещё в отключенном состоянии и network preload не добавлено, и у 90% остальных нормально тоже

я тоже это утверждаю :)
но это все субъективно
может это на стороне сайтеков вообще дообновлялись

но вот уход браузера в 100% который наблюдалсо в основном на гекко
явно очень точно совпал с переходом на 68.14.2

ту предзагрузку страниц которую ты упоминал я отключил
но это никак не помогло....

а проверять как работает старая версия руки недоходят....
теперь комп вытягивает только один браузер
запуск браузера тяжелый и долгий
а какойто один и так всегда нужный запущенный...

**Comment by NS-Clone** [comment]
> 20 olx к 20 других сайтов до этого открытым - не утекло

тут есть один момент
на каком проце ты оно запускаешь

а у нас у всех разное энергосбережение
например умя 9 p-state-ов
и на самом низком проц вообще выдает 800МГц
и так он работает пока не появитсо загрузка проца
а переключение частоты каждые 100мс
за это время можно пару сотен тыщ раз словить переполнение стека например
быстрые скроллы которые падают от переполнений стека на низкой частоте проца мы уже видели....

**Comment by NS-Clone** [comment]
> 20 olx к 20 других сайтов до этого открытым - не утекло

а у меня сегодня железно
полчаса лазения по олх-у
+2 гига утекает
закрывание браузера
не знаю сколько я ужо раз перезапускалсо....

...правадо это не проблема 68.14.2
с памятью стало туго давно

ну и умяж 1100 вкладок с предыдущей сессии
но они то не прогруженны
нахрена ж им жрать раму?
(хотя потом сто пудово окажетсо что они жрут и проц и раму)

**Comment by NS-Clone** [comment]
в процес хакере 2 процесса вообще набираеют по 30кб в сек
просто в простое

**Comment by NS-Clone** [comment]
с content process limit 3 и 4
потребление рамы и проца конское
но браузер не так вешаетсо

**Comment by Feodor2** [comment]
Для кого рисунки? Мне они ничём не говорят, я тоже могу напихать много разных рисунков.
Не было у меня проблем с олх и браузер сутками включен, комп отправляю в спячку когда должны выключить свет, вот сегодня не выключили - счастье то какое, но всё равно лучше перебздеть.

Опят таки что на чистой установке без 1000 вкладок, я думаю что все проблемы у тебя из-за этого.

**Comment by Al6bus** [comment]
> тут есть один момент на каком проце ты оно запускаешь
> 
> а у нас у всех разное энергосбережение например умя 9 p-state-ов и на самом низком проц вообще выдает 800МГц (да и для лета это критична Ж) и так он работает пока не появитсо загрузка проца а переключение частоты каждые 100мс за это время можно пару сотен тыщ раз словить переполнение стека например быстрые скроллы которые падают от переполнений стека на низкой частоте проца мы уже видели....
> 
> попробуй потестить уся на самой низкой возможной частоте проца можот можно дажо попробовать с вин15 там в энергопотреблении можно рулить p-state-ами через max min %

Пока что непонятно, что у тебя за комп и зачем ты занимаешься таким 314цом?

**Comment by Feodor2** [comment]
Okey one more thing that you may be missed.
When memory occupation of any process approaches 2gb it begins stutter, I think this is peculiarity of all firefoxes, nothing to do with this. I leave 3 process count so far, but further let reconsider about increase it.

**Comment by K4sum1** [comment]
This is not fixed in 68.14.3.

Limited content processes is not the issue, There seems to be some sort of RAM leak that occurs and gets cleared, then occurs and gets cleared again and again going by the video.

**Comment by NS-Clone** [comment]
but with
![2024-07-22_045017](https://github.com/user-attachments/assets/f88e5180-e368-407d-8de3-0302a4aa2275)
you can "use" browser again

**Comment by K4sum1** [comment]
I guess I could increase the processes. It being slow isn't really the issue, it's the freezing for me.

**Comment by K4sum1** [comment]
So I can get this to occur issue in minutes with no extensions enabled and just spam opening GitHub pages. Going to try fresh profile next.

**Comment by K4sum1** [comment]
Here is the issue occurring with a fresh profile in a few minutes, only settings changed are ctrl + tab to not be a shit, disabling hardware accel (idk why I did it happens with or without), and limiting to one content process all shown in the video.

https://files.catbox.moe/ys6boi.mp4

**Comment by K4sum1** [comment]
Here's Mypal 68.14.1 handling it like a champ, only a single significant freeze, nothing consistent, like how it was and should be.

https://files.catbox.moe/1f91sz.mp4

**Comment by K4sum1** [comment]
@Feodor2 This issue is even worse on my laptop. The browser gets unusable after a while. I'm going to be on my laptop for a month, I can't deal with this. I'm going to have to switch my main to r3dfox.

**Comment by K4sum1** [comment]
I can't say with 100% certainty, however it seems the issue is fixed in 68.14.5.

Doing the same test, the browser doesn't have these large lag spikes. I did notice there are some smaller lag spikes when typing that seemed to correspond to loading new ads that weren't apparent in the 68.14.1 test, however maybe I opened more tabs here than I did in previous tests or it could be related to changes with GitHub idk. They cleared up after a bit so I don't think it's a major issue.
